commands/peers/addpeer.go

```diff
@@ -26,112 +26,92 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var etcdConf peer.ETCDConfig
-	p := &peer.Peer{
-		ID:        uuid.Parse(rsp.UUID),
-		Name:      req.Name,
-		Addresses: req.Addresses,
-		MemberID:  0,
+	// If user hasn't specified peer name, use the name returned by remote
+	// peer which defaults to it's hostname.
+	if req.Name == "" {
+		req.Name = remotePeer.PeerName
 	}
 
-	// By default, req.Client is false. This means every new node added via
-	// add peer will be a member in etcd cluster and participate in
-	// consensus. TODO: This name "client" in the REST API should really be
-	// changed! May be to etcdproxy or just proxy ?
-	if req.Client == false {
-
-		// Adding a member is a two step process:
-		// 	1. Add the new member to the cluster via the members API. This is
-		//	   performed on this node i.e the one that just accepted peer add
-		//	   request from the user.
-		//	2. Start the new member on the target node (the new peer) with the new
-		//         cluster configuration, including a list of the updated members
-		//	   (existing members + the new member).
-
-		member, e := etcdmgmt.EtcdMemberAdd("http://" + req.Name + ":2380")
-		if e != nil {
-			log.WithFields(log.Fields{
-				"error":  e,
-				"member": req.Name,
-			}).Error("Failed to add member into etcd cluster")
-			rest.SendHTTPError(w, http.StatusInternalServerError, e.Error())
-			return
-		}
-
-		p.MemberID = member.ID
-		newName := p.Name
+	// Adding a member is a two step process:
+	// 	1. Add the new member to the cluster via the members API. This is
+	//	   performed on this node i.e the one that just accepted peer add
+	//	   request from the user.
+	//	2. Start the new member on the target node (the new peer) with the new
+	//         cluster configuration, including a list of the updated members
+	//	   (existing members + the new member).
 
+	newMember, e := etcdmgmt.EtcdMemberAdd("http://" + req.Address + ":2380")
+	if e != nil {
 		log.WithFields(log.Fields{
-			"New member ": newName,
-			"member Id ":  member.ID,
-		}).Info("New member added to the cluster")
-
-		mlist, e := etcdmgmt.EtcdMemberList()
-		if e != nil {
-			log.WithField("err", e).Error("Failed to list member in etcd cluster")
-			rest.SendHTTPError(w, http.StatusInternalServerError, e.Error())
-			return
-		}
+			"error":   e,
+			"uuid":    remotePeer.UUID,
+			"name":    req.Name,
+			"address": req.Address,
+		}).Error("Failed to add member to etcd cluster.")
+		rest.SendHTTPError(w, http.StatusInternalServerError, e.Error())
+		return
+	}
 
-		conf := []string{}
-		for _, memb := range mlist {
-			for _, u := range memb.PeerURLs {
-				n := memb.Name
-				if memb.ID == p.MemberID {
-					n = newName
-				}
-				conf = append(conf, fmt.Sprintf("%s=%s", n, u))
+	log.WithField("member-id", newMember.ID).Info("Added new member to etcd cluster")
+
+	mlist, e := etcdmgmt.EtcdMemberList()
+	if e != nil {
+		log.WithField("err", e).Error("Failed to list members in etcd cluster")
```

Comment:
The field key "err" is different from the one used above on line 64. Keep it consistent.
Also, when you have an error to be added to the log context, you could use `log.WithError`.